lib/eventsource/index.js

```diff
@@ -0,0 +1,448 @@
+
+    //   2. Let request be the EventSource object's request.
+    //   3. If the EventSource object's last event ID string is not the empty
+    //      string, then:
+    //      1. Let lastEventIDValue be the EventSource object's last event ID
+    //         string, encoded as UTF-8.
+    //      2. Set (`Last-Event-ID`, lastEventIDValue) in request's header
+    //         list.
+    if (this.#settings.lastEventId !== '') {
+      this.#request.headers.set('last-event-id', this.#settings.lastEventId, true)
+    }
+
+    //   4. Fetch request and process the response obtained in this fashion, if any, as described earlier in this section.
+    this.#connect()
+  }
+
+  /**
+   * Closes the connection, if any, and sets the readyState attribute to
+   * CLOSED.
+   */
+  close () {
+    webidl.brandCheck(this, EventSource)
+
+    if (this.#readyState === CLOSED) return
+    this.#readyState = CLOSED
+    clearTimeout(this.#settings.reconnectionTimer)
+    this.#controller.abort()
+
+    if (this.#request) {
+      this.#request = null
+    }
+  }
+
+  get onopen () {
+    webidl.brandCheck(this, EventSource)
+
+    return this.#events.open
+  }
+
+  set onopen (fn) {
+    webidl.brandCheck(this, EventSource)
+
+    if (this.#events.open) {
+      this.removeEventListener('open', this.#events.open)
+    }
+
+    if (typeof fn === 'function') {
+      this.#events.open = fn
+      this.addEventListener('open', fn)
+    } else {
+      this.#events.open = null
+    }
+  }
+
+  get onmessage () {
+    webidl.brandCheck(this, EventSource)
+
+    return this.#events.message
+  }
+
+  set onmessage (fn) {
+    webidl.brandCheck(this, EventSource)
+
+    if (this.#events.message) {
+      this.removeEventListener('message', this.#events.message)
+    }
+
+    if (typeof fn === 'function') {
+      this.#events.message = fn
+      this.addEventListener('message', fn)
+    } else {
+      this.#events.message = null
+    }
+  }
+
+  get onerror () {
+    webidl.brandCheck(this, EventSource)
+
+    return this.#events.error
+  }
+
+  set onerror (fn) {
+    webidl.brandCheck(this, EventSource)
+
+    if (this.#events.error) {
+      this.removeEventListener('error', this.#events.error)
+    }
+
+    if (typeof fn === 'function') {
+      this.#events.error = fn
+      this.addEventListener('error', fn)
+    } else {
+      this.#events.error = null
+    }
+  }
+}
+
+Object.defineProperties(EventSource, {
+  CONNECTING: {
+    __proto__: null,
+    configurable: false,
+    enumerable: true,
+    value: CONNECTING,
+    writable: false
+  },
+  OPEN: {
+    __proto__: null,
+    configurable: false,
+    enumerable: true,
+    value: OPEN,
+    writable: false
+  },
+  CLOSED: {
+    __proto__: null,
+    configurable: false,
+    enumerable: true,
+    value: CLOSED,
+    writable: false
+  }
+})
+
+EventSource.prototype.CONNECTING = CONNECTING
+EventSource.prototype.OPEN = OPEN
+EventSource.prototype.CLOSED = CLOSED
```

Comment:
I believe these similarly have to be defined with Object.defineProperties making them not configurable & writable